tests/suites/test_suite_psa_crypto.function

```diff
@@ -6897,19 +6897,27 @@ exit:
 
 /* BEGIN_CASE */
 void derive_output( int alg_arg,
-                    int step1_arg, data_t *input1,
-                    int step2_arg, data_t *input2,
-                    int step3_arg, data_t *input3,
+                    int step1_arg, data_t *input1, int expected_status_arg1,
```

Comment:
Why add support for bad inputs in `derive_output`? Testing bad input sequences should be the job of `derive_input`. What's missing from `derive_input`? Why add it to `derive_output`?

The key derivation test functions are already quite messy as it is and I don't like increasing their complexity.

Reply:
I can't speak for Przemek, but as far as I'm concerned, when I touch an area I'm not familiar with, I often find it non-trivial to understand what the purpose of each test function is and how they complement each other.

I think in general when writing tests we should write more documentation about the general strategy, rather than expect future readers to infer it from the list of existing test functions. (I've been saying that a few times now.)

Reply:
Yes, it is hard for newbies to understand what is where. A lot of code is provided in the test files and usually without any information or testing strategy.
While working on `PSK-to-MS` I found the test function that was closest to my needs and extended it to test the new code including negative test cases. Also added comments in `.data` file to give some extra details on testing.

Need to take a closer look into this. 
Also I can try to clean this and move negative test cases for inputs to `derive_input` test function.